src/test/java/com/google/research/bleth/simulator/IAgentTest.java

```diff
@@ -0,0 +1,262 @@
+package com.google.research.bleth.simulator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public abstract class IAgentTest {
```

Comment:
Same as with AwakenessStrategyTest - having an abstract test class for an interface is again the right choice.